test/Microsoft.AspNetCore.Cors.Test/CorsPolicyBuilderTests.cs

```diff
@@ -299,5 +299,85 @@ public void DisallowCredential_SetsSupportsCredentials_ToFalse()
             var corsPolicy = builder.Build();
             Assert.False(corsPolicy.SupportsCredentials);
         }
+
+        [Theory]
+        [InlineData(@"Some-String", "some-string")]
+        [InlineData(@"x:\\Test", "x:\\test")]
+        [InlineData("FTP://Some-url", "ftp://some-url")]
+        public void GetNormalizedOrigin_ReturnsLowerCasedValue_IfStringIsNotHttpOrHttpsUrl(string origin, string expected)
+        {
+            // Act
+            var normalizedOrigin = CorsPolicyBuilder.GetNormalizedOrigin(origin);
+
+            // Assert
+            Assert.Equal(expected, normalizedOrigin);
+        }
+
+        [Fact]
+        public void GetNormalizedOrigin_DoesNotAddPort_IfUriDoesNotSpecifyOne()
+        {
+            // Arrange
+            var origin = "http://www.example.com";
+
+            // Act
+            var normalizedOrigin = CorsPolicyBuilder.GetNormalizedOrigin(origin);
+
+            // Assert
+            Assert.Equal(origin, normalizedOrigin);
+        }
+
+        [Fact]
+        public void GetNormalizedOrigin_LowerCasesScheme()
+        {
+            // Arrange
+            var origin = "HTTP://www.example.com";
+
+            // Act
+            var normalizedOrigin = CorsPolicyBuilder.GetNormalizedOrigin(origin);
+
+            // Assert
+            Assert.Equal("http://www.example.com", normalizedOrigin);
+        }
+
+        [Fact]
+        public void GetNormalizedOrigin_LowerCasesHost()
+        {
+            // Arrange
+            var origin = "http://www.Example.Com";
+
+            // Act
+            var normalizedOrigin = CorsPolicyBuilder.GetNormalizedOrigin(origin);
+
+            // Assert
+            Assert.Equal("http://www.example.com", normalizedOrigin);
+        }
+
+        [Theory]
+        [InlineData("http://www.Example.com:80", "http://www.example.com:80")]
+        [InlineData("https://www.Example.com:8080", "https://www.example.com:8080")]
+        public void GetNormalizedOrigin_PreservesPort_ForNonIdnHosts(string origin, string expected)
```

Comment:
Updated test name

Reply:
Having different default port behavior for IDNs and non-IDNS doesn't sound like it's actually going to work. From your description above it sounds like dropping the default port is what browsers expect.

Reply:
My concern was that it's a really large browser matrix to test to be 100% certain. It seemed less problematic with Ids since they weren't working to start with. But it'd be more frustrating if we broke it for apps \ browsers that worked correctly.

Reply:
You need to explain all of this normalization in the doc comments on WithOrigins.

Reply:
Added docs